src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2363,7 +2365,15 @@ public void Disconnect(bool reuseSocket)
         //    int - Return code from async Connect, 0 for success, SocketError.NotConnected otherwise
         public void EndConnect(IAsyncResult asyncResult)
         {
-            ThrowIfDisposed();
+            if (Disposed)
+            {
+                if (SocketsTelemetry.Log.IsEnabled() && asyncResult is not MultipleAddressConnectAsyncResult)
```

Comment:
Why is this conditioned on it not being a MultipleAddressConnectAsyncResult?

Reply:
MultipleConnect spawns its own 'child' connects that already log Start/Failed/Stop events. We don't log Start for the 'parent' operation to avoid duplicate events, so we avoid logging stop here as well.